springfox-core/src/main/java/springfox/documentation/builders/OperationBuilder.java

```diff
@@ -232,9 +233,20 @@ public OperationBuilder tags(Set<String> tags) {
     return this;
   }
 
+  /**
+   * Sets the nickname for the operation
+   *
+   * @param nickname - a string for the nickname
+   * @return this
+   */
+  public OperationBuilder nickname(String nickname) {
+      this.nickname = defaultIfAbsent(nickname, this.nickname);
+      return this;
+  }
+
   public Operation build() {
     return new Operation(method, summary, notes, responseModel, uniqueId, position, tags, produces,
-        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden);
+        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden, nickname);
```

Comment:
@who thanks for contributing this!!

Referencing [the related issue](https://github.com/springfox/springfox/issues/659)

I would think that ideally we would have the following:
- Currently we do not include the operationId by default - this should remain.
- Provide a way for the user to configure and enable operationId generation via [Docket](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/plugins/Docket.java#L63-63)
```java
. operationIdGenerator(someBeanOrFunctionThatPopulatesUniqueId())
```
- We should include a default `OperationIdGenerator` which generates a sensible id by default and would have the behaviour described in no.2 above (override it when a nickname is supplied in @ApiOperation). Users would enable it as follows:
```java
. operationIdGenerator(new OperationIdGenerator())
```